rust/lance-index/src/scalar/inverted/index.rs

```diff
@@ -229,17 +229,38 @@ impl ScalarIndex for InvertedIndex {
     where
         Self: Sized,
     {
-        let token_reader = store.open_index_file(TOKENS_FILE).await?;
-        let invert_list_reader = store.open_index_file(INVERT_LIST_FILE).await?;
-        let docs_reader = store.open_index_file(DOCS_FILE).await?;
+        let tokens_fut = tokio::spawn({
+            let store = store.clone();
+            async move {
+                let token_reader = store.open_index_file(TOKENS_FILE).await?;
+                let tokens = TokenSet::load(token_reader).await?;
+                Result::Ok(tokens)
+            }
+        });
+        let invert_list_fut = tokio::spawn({
+            let store = store.clone();
+            async move {
+                let invert_list_reader = store.open_index_file(INVERT_LIST_FILE).await?;
+                let invert_list = InvertedListReader::new(invert_list_reader)?;
+                Result::Ok(Arc::new(invert_list))
+            }
+        });
+        let docs_fut = tokio::spawn({
+            let store = store.clone();
+            async move {
+                let docs_reader = store.open_index_file(DOCS_FILE).await?;
+                let docs = DocSet::load(docs_reader).await?;
+                Result::Ok(docs)
+            }
+        });
 
-        let tokens = TokenSet::load(token_reader).await?;
-        let inverted_list = InvertedListReader::new(invert_list_reader)?;
-        let docs = DocSet::load(docs_reader).await?;
+        let tokens = tokens_fut.await??;
+        let inverted_list = invert_list_fut.await??;
+        let docs = docs_fut.await??;
```

Comment:
You can await multiple futures at the same time with `try_join!()`:
```suggestion
        let (tokens, inverted_list, docs) = try_join!(tokens_fut, invert_list_fut, docs_fut)?;
```

This has the upside that it will fail on the first failure of any of the three.

Reply:
Yes your solution is what I tried, then I encountered that "`FnOnce` is not general enough" error, so decided to use `spawn`, with `spawn()` then I think no need to use `try_join!` because these tasks are running concurrently once spawned.

It's a good idea to split the IO/CPU operations, now the `load` methods mixed them

Reply:
Okay this is fine then.